internal/kms/kms.go

```diff
@@ -390,29 +407,34 @@ func (k *Kms) loadDek(ctx context.Context, scopeId string, purpose KeyPurpose, r
 		return nil, errors.New(ctx, errors.InvalidParameter, op, "unknown or invalid DEK purpose specified")
 	}
 	if err != nil {
-		return nil, errors.Wrap(ctx, err, op, errors.WithMsg(fmt.Sprintf("error looking up %s key versions for scope %s with key ID %s", purpose.String(), scopeId, rootWrapper.KeyID())))
+		return nil, errors.Wrap(ctx, err, op, errors.WithMsg(fmt.Sprintf("error looking up %s key versions for scope %s", purpose.String(), scopeId)))
 	}
 	if len(keyVersions) == 0 {
 		return nil, errors.New(ctx, errors.KeyNotFound, op, fmt.Sprintf("no %s key versions found for scope %s", purpose.String(), scopeId))
 	}
 
-	var multi *multiwrapper.MultiWrapper
+	var pooled *multi.PooledWrapper
 	for i, keyVersion := range keyVersions {
-		wrapper := aead.NewWrapper(nil)
-		if _, err := wrapper.SetConfig(map[string]string{
-			"key_id": keyVersion.GetPrivateId(),
-		}); err != nil {
+		var err error
+		wrapper := aead.NewWrapper()
+		if _, err = wrapper.SetConfig(ctx, wrapping.WithKeyId(keyVersion.GetPrivateId())); err != nil {
 			return nil, errors.Wrap(ctx, err, op, errors.WithMsg(fmt.Sprintf("error setting config on aead %s wrapper in scope %s", purpose.String(), scopeId)))
 		}
-		if err := wrapper.SetAESGCMKeyBytes(keyVersion.GetKey()); err != nil {
+		if err = wrapper.SetAesGcmKeyBytes(keyVersion.GetKey()); err != nil {
 			return nil, errors.Wrap(ctx, err, op, errors.WithMsg(fmt.Sprintf("error setting key bytes on aead %s wrapper in scope %s", purpose.String(), scopeId)))
 		}
 		if i == 0 {
-			multi = multiwrapper.NewMultiWrapper(wrapper)
+			pooled, err = multi.NewPooledWrapper(ctx, wrapper)
+			if err != nil {
+				return nil, errors.Wrap(ctx, err, op, errors.WithMsg(fmt.Sprintf("error getting %s multiwrapper for key version 0 in scope %s", purpose.String(), scopeId)))
```

Comment:
same question about multi vs pooled in the error msg.

Reply:
Updated